doc/source/flexibility/compressed_columns.rst

```diff
@@ -5,54 +5,69 @@ Compressed columns with dictionaries
-data types are created. Compression is done by using the ``zlib`` library.
-
-Additionally it is possible to pre-define a set of strings for each compressed
-column to achieve a better compression ratio on a relatively small individual
+compression feature. It is a data type modifier, independent from user-level SQL
+and |InnoDB| data compression, that causes the data stored in the column to be
+compressed on writing to storage and decompressed on reading. For all other
+purposes, the data type is identical to the one without the modifier, i.e. no
+new data types are created. Compression is done by using the ``zlib`` library.
+
+Additionally, it is possible to pre-define a set of strings for each compressed
+column to achieve a better compression ratio on relatively small individual
 data items.
 
 This feature provides:
 
 * a better compression ratio for text data which consist of a large number of
   predefined words (e.g. JSON or XML) using compression methods with static
-  dictionaries,
-* in contrast to |InnoDB| row compression method it provides a way to select
-  which columns in the table must be compressed.
-
+  dictionaries
+* a way to select columns in the table to compress (in contrast to the |InnoDB|
+  row compression method)
+  
 This feature is based on a patch provided by Weixiang Zhai.
 
 Specifications
 ==============
 
-The feature is limited to InnoDB/XtraDB storage engine and to:
+The feature is limited to InnoDB/XtraDB storage engine and to columns of the
+following data types:
+
+* ``BLOB`` (including ``TINYBLOB``, ``MEDIUMBLOB``, ``LONGBLOG``)
 
-* ``BLOB`` (including ``TINYBLOB``, ``MEDIUMBLOB``, ``LONGBLOG``),
+* ``TEXT`` (including ``LONGTEXT``)
 
-* ``TEXT`` (including ``LONGTEXT``),
+* ``VARCHAR`` (including ``NATIONAL VARCHAR``)
 
-* ``VARCHAR`` (including ``NATIONAL VARCHAR``), and
+* ``VARBINARY``
 
-* ``VARBINARY`` columns.
+* ``JSON``
 
 The syntax to declare a compressed column is using an extension of an existing
 ``COLUMN_FORMAT`` modifier: ``COLUMN_FORMAT COMPRESSED``. If this modifier is
-applied on an unsupported column type or storage engine, an error is returned.
+applied to an unsupported column type or storage engine, an error is returned.
 
 The compression can be specified:
 
-* at the table create time:
+* when creating a table:
   ``CREATE TABLE ... (..., foo BLOB COLUMN_FORMAT COMPRESSED, ...);``
-
-* or a table can be modified to compress/decompress a column later:
+  
+* when altering a table and modifying a column to the compressed format:
   ``ALTER TABLE ... MODIFY [COLUMN] ... COLUMN_FORMAT COMPRESSED``, or
   ``ALTER TABLE ... CHANGE [COLUMN] ... COLUMN_FORMAT COMPRESSED``.
 
-To decompress a column, a ``COLUMN_FORMAT`` value any other than ``COMPRESSED``
-can be specified: ``FIXED``, ``DYNAMIC``, or ``DEFAULT``. If there is a column
+Unlike Oracle MySQL, compression is applicable to generated columns, both
+virtual and stored. Use this syntax extension as follows:
```

Comment:
is applicable to generated STORED columns only (it does not make sense for generated VIRTUAL columns as their values are calculated on the fly)